Makefile

```diff
@@ -47,72 +30,17 @@ AD_SERVER_SOURCES=\
+BUILDFILE_SERVER=$(shell $(call _BUILD_KP,server,default,--print-buildfile))
+BUILDFILE_NIGHTLY_SERVER=$(shell $(call _BUILD_KP,server,nightly,--print-buildfile))
+BUILDFILE_AD_SERVER=$(shell $(call _BUILD_KP,ad-server,default,--print-buildfile))
+BUILDFILE_NIGHTLY_AD_SERVER=$(shell $(call _BUILD_KP,ad-server,nightly,--print-buildfile))
+BUILDFILE_CLIENT=$(shell $(call _BUILD_KP,client,default,--print-buildfile))
+BUILDFILE_TOOLBOX=$(shell $(call _BUILD_KP,toolbox,default,--print-buildfile))
 
 REPO_BASE=quay.io/samba.org/
-SERVER_REPO_NAME=$(REPO_BASE)$(SERVER_NAME)
-NIGHTLY_SERVER_REPO_NAME=$(REPO_BASE)$(NIGHTLY_SERVER_NAME)
-AD_SERVER_REPO_NAME=$(REPO_BASE)$(AD_SERVER_NAME)
-NIGHTLY_AD_SERVER_REPO_NAME=$(REPO_BASE)$(NIGHTLY_AD_SERVER_NAME)
-CLIENT_REPO_NAME=$(REPO_BASE)$(CLIENT_NAME)
-NIGHTLY_CLIENT_REPO_NAME=$(REPO_BASE)$(NIGHTLY_CLIENT_NAME)
-TOOLBOX_REPO_NAME=$(REPO_BASE)$(TOOLBOX_NAME)
-NIGHTLY_TOOLBOX_REPO_NAME=$(REPO_BASE)$(NIGHTLY_TOOLBOX_NAME)
 
-BUILDFILE_PREFIX=.build
-BUILDFILE_SERVER=$(BUILDFILE_PREFIX).$(OS_PREFIX)server
-BUILDFILE_NIGHTLY_SERVER=$(BUILDFILE_PREFIX).$(OS_PREFIX)nightly-server
-BUILDFILE_AD_SERVER=$(BUILDFILE_PREFIX).$(OS_PREFIX)ad-server
-BUILDFILE_NIGHTLY_AD_SERVER=$(BUILDFILE_PREFIX).$(OS_PREFIX)nightly-ad-server
-BUILDFILE_CLIENT=$(BUILDFILE_PREFIX).$(OS_PREFIX)client
-BUILDFILE_NIGHTLY_CLIENT=$(BUILDFILE_PREFIX).$(OS_PREFIX)nightly-client
-BUILDFILE_TOOLBOX=$(BUILDFILE_PREFIX).$(OS_PREFIX)toolbox
-BUILDFILE_NIGHTLY_TOOLBOX=$(BUILDFILE_PREFIX).$(OS_PREFIX)nightly-toolbox
-
-HOST_ARCH:=$(shell arch)
-HOST_ARCH:=$(subst x86_64,amd64,$(HOST_ARCH))
-HOST_ARCH:=$(subst aarch64,arm64,$(HOST_ARCH))
-
-# build_fqin is a function macro for building a "Fully Qualified Image Name".
-# Usage: $(call build_fqin,<base-name>,<pkg-source>,<os-name>,<arch>,[<extra>])
-#   base-name: the last part of the repo name eg. 'samba-server'
-#   pkg-source: source for samba packages (default or nightly)
-#   os-name: base os name
-#   arch: architecture of image (amd64, arm64, etc.)
-#   extra: (optional) an additional unique suffix for the tag
-#          typically meant for use by devs building custom images
-build_fqin=$(REPO_BASE)$(1):$(2)-$(3)-$(4)$(if $(5),-$(5))
-
-# get_imagename is a function macro for getting only the base image name
-# without the tag part.
-# Usage: $(call get_imagename,<image-name>)
-get_imagename=$(firstword $(subst :, ,$1))
-
-# get_pkgsource is a function macro that, given an images name returns
-# the name of the package source. Currently only understands the
-# difference between default (os packages) and nightly (SIT packages).
-# Usage: $(call, get_pkgsource,<image-name>)
-get_pkgsource=$(if $(findstring nightly,$1),nightly,default)
+_BUILD_KP=$(BUILD_IMAGE) $(if $(CONTAINER_CMD),--container-engine=$(CONTAINER_CMD)) $(BI_PREFIX_ARGS) --kind=$1 --package-source=$2 --distro-base=$(SRC_OS_NAME)  --repo-base=$(REPO_BASE) $(if $(BUILD_ARCH),--arch=$(BUILD_ARCH)) $3
```

Comment:
Line too long. Consider split into multi-lines

Reply:
It's a good idea but make can be weirdly sensitive to line continuations with `\` .  If you don't mind I'd like to leave it as-is for now. We can always come back to it in a future patch. Is that OK?

Reply:
Sure; it is just an idea.